### ISSUE TITLE
Staff Status flag allows access to public datasets

### ISSUE THREAD

**Issue by RichardPBerry** [mytardis/mytardis#789]
Don't think this is intended behaviour, but found out that turning on the "Staff Status" flag for a user allows that user (with no other permissions assigned) to edit the properties of an experiment with public access set to anything other than private. This includes giving themselves full owner rights over a publicly listed experiment.

**Comment by grischa** [comment]
Where can you do those edits? Is it in the admin interface or the front page web interface?

**Comment by RichardPBerry** [comment]
@grischa through the UI on the Sharing panel.

**Comment by grischa** [comment]
Thanks! I'll investigate

**Comment by grischa** [comment]
@pansapiens implemented it this way last year.
I suppose we need to document what "Staff" means. What did you intend "Staff" to mean @RichardPBerry and @pansapiens ?

**Comment by RichardPBerry** [comment]
Ok, thanks. Only happened on this by accident when testing the publication forms. Would have expected 'Staff' to allow access to the admin site which is what the text next to the checkbox says.

**Comment by pansapiens** [comment]
We would be best to follow the Django convention, which is to make 'staff' allow access to the admin site, but respecting existing user permissions.

In the tests I'd done, I had concluded that `is_staff` also implies broad edit access via `/admin`. This is because by default, new users are registered with the `tardis_acls.change_objectacl` permission, which when given `/admin` access via 'staff' allows them to change the ObjectACLs for _any_ experiment. Following this logic, it seemed reasonable to also allow this in the web UI as an added convenience.

This may not always be the case, and in hindsight seems like at best an unexpected (and undocumented!) deviation from convention, at worst a potential security/privacy issue - I've made a PR to revert this behaviour: https://github.com/mytardis/mytardis/pull/807

It's worth noting that Django `/admin` in MyTardis currently doesn’t respect ObjectACLs, so the only context for currently using the 'Staff' flag in MyTardis is if you wanted to give an administrator or facility manager access to add/edit certain model types (eg all `Instruments`), without giving access to other models (eg modify any `Users` or `Facilities`). 'Staff' access isn't really suitable for a multi-Facility installation since one facility manager with `staff = True` and `tardis_acls.change_experiment` would be able to modify experiments generated by another facility. So I'd suggest the use case of the `staff` flag in MyTardis is currently pretty restricted in utility and should mostly be avoided without carefully testing the implications.

----
Some additional notes - I did a few manual tests to determine current (pre-PR 807) behaviour:

If a user is `staff = True` with no additional user permissions or groups, created via Django `/admin`:
	• They can access an empty admin page but can't view (or edit) any data. 
	• They _cannot_ edit any public experiments/datasets via the MyTardis web UI (there is no edit button).

If a user is created via the MyTardis web UI (eg registration or first login) they receive a set of default user permissions that allow them to modify ObjectACLs, add and change experiments etc. The MyTardis web UI further restricts their actions to experiments they own via ObjectACLs. If this user is then given `staff = True` and access to `/admin` they can:
	• They modify anyones `Experiment`, `Dataset`, `User  authentication` or `ObjectACL` in the database via `/admin`.
	• They can modify sharing / ownership of any Experiment via the web UI (<-- this is the functionality PR https://github.com/mytardis/mytardis/pull/807 with remove).


If a user is `staff = True` with only the `tardis_acls.change_experiment` permission:
	• They can access admin and change _all_ `Experiments` - this is _not_ restricted to experiments where they have owership/edit permission via `ObjectACL`s.
	• They _cannot_ edit any public experiments/datasets via the MyTardis web UI.

If a user is `staff = True` with ownership of an experiment via ObjectACLs, with no other user permissions:
	• They can change sharing settings of the experiment, and edit it's metadata in the MyTardis web UI (expected behaviour).
	• They get and empty admin page. They _can't_ edit the experiment or associated `ObjectACL`s via Django admin - Django admin isn’t customized to recognise MyTardis `ObjectACL`s.

If the user is a member of a group they inherit permissions and `ObjectACL`-based access from the group, as expected.
